com.ibm.streamsx.utility/com.ibm.streamsx.utility/native.function/utility.cpp

```diff
@@ -104,4 +105,10 @@ namespace com { namespace ibm { namespace streamsx { namespace utility {
         if (rc<0) THROW (SPLRuntimeOperator, "could not set processor affinity to CPUs"
             << ", " << strerror(errno));
     }
+
+    void setThreadName(SPL::rstring name)
+    {
+        pthread_setname_np(pthread_self(), name.c_str());
```

Comment:
Maybe it wasn't RHEL6 - it might have been a Power system, or maybe even SUSE. I broke a product build by delivering code with `pthread_setname_np`. (It worked on my development system.)

For error checking, that sounds good. I did something similar in the product.